src/program.fs

```diff
@@ -165,26 +176,22 @@ module Program =
                     while not terminated && Option.isSome nextMsg do
                         let msg = nextMsg.Value
                         if toTerminate msg then
+                            Subs.stop program.onError subs
                             terminate state
                             terminated <- true
                         else                        
                             let (model',cmd') = program.update msg state
                             program.setState model' dispatch'
                             cmd' |> Cmd.exec (fun ex -> program.onError (sprintf "Error handling the message: %A" msg, ex)) dispatch'
+                            subs <- program.subscribe model' |> Subs.recalc subs |> Subs.change program.onError dispatch'
```

Comment:
I'm curious about the placement, should it not follow `update` directly?

Reply:
If you mean readability, yeah it would have been better to keep decisions above and effects below. I reckon I was trying to keep the footprint minimal. If you mean side effect order, my thought was principle of least surprise for the user. init/update is obviously run first, since the new model is needed for subscribe. So their effects are also run in that order.

Reply:
See what you think now.

Reply:
Right, I don't have a strong preference atm, just trying to figure the intent. To me `update` and `subscribe` are about generating the inputs and `setState` and `exec` are about affecting the change, which is why I thought I'd place them together.